### ISSUE TITLE
Using `--format group` filters out some outdated packages.

### ISSUE THREAD

**Issue by gilbarbara** [raineorshine/npm-check-updates#1158]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14`

---

Hey @raineorshine!

Using the ncu command with `--format group` filters out some outdated packages.

```sh
ncu -ui --format group
Upgrading /Users/gilbarbara/code/---/package.json
[====================] 22/22 100%

Patch   Backwards-compatible bug fixes
❯ ◉ @types/node                ^18.0.0  →  ^18.0.1
  ◉ @types/react-dom           ^18.0.5  →  ^18.0.6
```

```sh
ncu -ui           
Upgrading /Users/gilbarbara/code/---/package.json
[====================] 22/22 100%

? Choose which packages to update › Space to deselect. Enter to upgrade. 
❯ ◉ @gilbarbara/components      ^0.4.3  →   ^0.4.4
  ◉ @gilbarbara/eslint-config   ^0.2.4  →   ^0.2.5
  ◉ @types/node                ^18.0.0  →  ^18.0.1
  ◉ @types/react-dom           ^18.0.5  →  ^18.0.6
  ◉ vite-plugin-pwa            ^0.12.0  →  ^0.12.2
```

## Steps to Reproduce

Dependencies:

```json
{
  "dependencies": {
    "@emotion/react": "^11.9.3",
    "@emotion/styled": "^11.9.3",
    "@gilbarbara/components": "^0.4.3",
    "immer": "^9.0.15",
    "is-lite": "^0.8.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-use": "^17.4.0"
  },
  "devDependencies": {
    "@gilbarbara/eslint-config": "^0.2.4",
    "@gilbarbara/prettier-config": "^0.1.0",
    "@gilbarbara/tsconfig": "^0.1.1",
    "@gilbarbara/types": "^0.1.1",
    "@types/node": "^18.0.0",
    "@types/react": "^18.0.14",
    "@types/react-dom": "^18.0.5",
    "@vitejs/plugin-react": "^1.3.2",
    "type-fest": "^2.16.0",
    "typescript": "^4.7.4",
    "vite": "^2.9.13",
    "vite-plugin-environment": "^1.1.1",
    "vite-plugin-pwa": "^0.12.0",
    "vite-tsconfig-paths": "^3.5.0"
  }
}
```

Steps:

Just run `ncu -ui --format group`

## Current Behavior
Some packages aren't shown with `--format group`

## Expected Behavior
All outdated packages should be listed.

**Comment by raineorshine** [comment]
Thank you! 

That's what I get for making an overly loose type 😅

Fixed and published in `v15.0.4`

**Comment by gilbarbara** [comment]
Awesome! 👏🏻

However, `Non-Server` is a bit weird since the versions adhere to semver, and the 0.x.x pattern is a common practice that a lot of packages use.
The only real difference is that **minor** updates can have breaking changes, but **patch** works as usual.

The way it is this block feels like an else statement. :)

**Comment by raineorshine** [comment]
Yes, you're right, that's not the best name for it.

Semver itself refers to it as [major version zero](https://semver.org/#spec-item-4).

Other options:

- Major version zero (matches the semver spec, but is a little wordy)
- Pre v1
- 0.x

> The only real difference is that **minor** updates can have breaking changes, but **patch** works as usual.

Technically, **patch** updates in major version zero can be breaking, too, though common practice may be to only use them for non-breaking updates.

**Comment by raineorshine** [comment]
I have renamed it to **Major version zero** and published to `v15.0.5`. Thanks for the feedback!

**Comment by gilbarbara** [comment]
hey,

> Technically, patch updates in major version zero can be breaking, too, though common practice may be to only use them for non-breaking updates.

Yeah, 0.x is unstable after all.

Thanks
